lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionSourceCode.cpp

```diff
@@ -329,16 +329,37 @@ __builtin_logger_initialize()
 
   assert(!playground && !repl && "Playground/REPL mode not expected");
 
+  auto path_literal = [](const char *path) -> std::string {
+    llvm::SmallString<261> buffer{path};
+    llvm::sys::path::native(buffer);
+
+    std::string literal = buffer.str().str();
+
+    std::string escaped;
+    escaped.reserve(2 * literal.size());
+
+    for (std::string::size_type start = 0, end = 0;
+         end = literal.find('\\', start); start = end + 1) {
+      escaped.append(literal, start, end - start);
+      if (end == std::string::npos)
+        break;
+      escaped.append("\\\\");
+    }
+
+    return escaped;
+  };
+
   if (pound_file && pound_line) {
     fixed_text.Printf("#sourceLocation(file: \"%s\", line: %u)\n%s\n",
-                      pound_file, pound_line, orig_text);
+                      path_literal(pound_file).c_str(), pound_line, orig_text);
```

Comment:
would this work too?
```
std::string s;
{
llvm::raw_string_ostream os(s);
llvm::printEscapedString(pound_file, os);
}

s.c_str()
```

Reply:
hmm, I'm not sure if it would potentially break on some unexpected input.  But, it is simpler, and does escape the `\` properly AFAICT.  `"` is not a valid character, but nearly everything else might be mapped properly?  I think I kinda prefer that 🤔.  Let me just verify that it does work and I'll upload a new version.

Reply:
Seems to work locally?  Let's go with that.